src/components/SortByForm.jsx

```diff
@@ -0,0 +1,37 @@
+const SortByForm = ({ setSort, setOrder, setSearchParams, query }) => {
+  const changeHandler = event => {
+    const sortingValues = event.target.value.split(',');
+
+    if (query) {
+      setSearchParams({
+        topic: query,
+        sort_by: sortingValues[0],
+        order: sortingValues[1],
+      });
+    } else {
+      setSearchParams({
+        sort_by: sortingValues[0],
+        order: sortingValues[1],
+      });
+    }
+    setSort(sortingValues[0]);
+    setOrder(sortingValues[1]);
+  };
+
+  return (
+    <form>
+      <label htmlFor="">Sort by</label>
+      <select onChange={changeHandler}>
+        <option value="votes,DESC">Most Popular </option>
+        <option value="created_at,DESC">Most Recent Articles</option>
+        <option value="created_at,ASC">Oldest Articles</option>
+        <option value="author,ASC">Author (Ascending)</option>
+        <option value="author,DESC">Author (Descending)</option>
+        <option value="title,ASC">Title (Ascending)</option>
+        <option value="title,DESC">Title (Descending)</option>
+      </select>
+    </form>
+  );
+};
+
```

Comment:
Nice implementation of sorting and it seems to work flawlessly 👏 , nice one
However from a UX perspective it would be preferable for the sorting (`order`) selection to be separate from the thing the articles are sorted by (the `sort_by`) rather than a drop down menu with all the combinations.
This is by no means mandatory to implement but worth keeping in mind if you wanted to improve the UX